src/smoosh/test/smoosh_tests.erl

```diff
@@ -83,12 +83,15 @@ should_enqueue(ChannelType, DbName) ->
 
 should_persist_queue(ChannelType, DbName) ->
     ?_test(begin
+        {ok, ChannelPid} = smoosh_server:get_channel(ChannelType),
         ok = grow_db_file(DbName, 300),
         ok = wait_enqueue(ChannelType, DbName),
-        Q = channel_queue(ChannelType),
+        ok = smoosh_channel:persist(ChannelPid),
```

Comment:
This line defeats the purpose of this test. My understanding of the goal of the test is to make sure all plumbing is correct and we get to the place where smoosh persists the state. If we call the function from the test directly we no longer test the plumbing.

Reply:
It was more intended to test the functionality of `smoosh_priority_queue:persist/1` and the read-in on restart. Not necessarily the initial write that occurs on restart. Testing the initial write has proven to be difficult due to the racing that occurs between the test and `smoosh_channel`. Any ideas on solutions to prevent a race from occurring?